src/azure.ad/basicauth_inbound_handler.bal

```diff
@@ -0,0 +1,111 @@
+import ballerina/http;
+import ballerina/mime;
+import ballerina/lang.'string as sutils;
+
+# Creates an inbound auth handler which authenticates users against the active directory based on basic auth headers.
+# 
+# + tenantId - Tenant ID for the active directory
+# + clientId - Client ID for the client credentials grant authentication
+# + clientSecret - Client secret for the client credentials grant authentication
+# + scopes - Scope(s) of the access request
+# + return - Inbound basic auth handler
+public function getAzureADInboundBasicAuthHandler(string tenantId, string clientId, string clientSecret, string|string[] scopes = ["https://graph.microsoft.com/.default"]) returns http:BasicAuthHandler { 
+    InboundAzureADUserAuthenticatorProvider userAuthProvider = new(tenantId, clientId, clientSecret, scopes);
+    http:BasicAuthHandler basicAuthHandler = new(userAuthProvider);
+    return basicAuthHandler;
+}
+
+# An inbound authentication provider which validates used against the active directory when basis auth token is provided.
+public type InboundAzureADUserAuthenticatorProvider object {
+
+    *auth:InboundAuthProvider;
+
+    private string clientId;
+    private string clientSecret;
+    private string 'resource;
+    private string|string[] scopes;
+    private http:Client msOAuth2Client;
+
+    # Provides authentication based on the provided introspection configurations.
+    #
+    # + tenantId - Tenant ID for the active directory
+    # + clientId - Client ID for the client credentials grant authentication
+    # + clientSecret - Client secret for the client credentials grant authentication
+    # + scopes - Scope(s) of the access request
+    # + 'resource - The resource which the authentication occurrs
+    public function __init(string tenantId, string clientId, string clientSecret, string|string[] scopes, string 'resource = "https://graph.microsoft.com") {
+        self.clientId = clientId;
+        self.clientSecret = clientSecret;
+        self.'resource = 'resource;
+        self.scopes = scopes;
+        self.msOAuth2Client = new(string `https://login.microsoftonline.com/${tenantId}/oauth2`);
+    }
+
+    # Use OAuth2 password grant type to authenticate user based on their basic auth token.
+    # 
+    # + credential - The user's basic auth token
+    # + return - `true` if authenticated, `false` if not authenticated, `error` if error occurred while authenticating
+    public function authenticate(string credential) returns boolean|auth:Error {
+        // Format scopes into a string.
+        string|string[] oAuthScopes = self.scopes;
+        string oAuthScopesFormatted;
+        if (oAuthScopes is string) {
+            oAuthScopesFormatted = oAuthScopes;
+        } else {
+            oAuthScopesFormatted = sutils:'join(" ", ...oAuthScopes);
+        }
+
+        [string, string] [username, password] = check auth:extractUsernameAndPassword(credential);
+
+        string payload = "grant_type=password&" +
```

Comment:
Can we use `oauth2:OutboundOAuth2Provider` with password grant type config here?

Reply:
I didn't use it because it is a single request and there is no need to use caching the tokens and expiry time. Would it still make sense to use it ?

Reply:
Then its fine to keep as this.